test/e2e/apimachinery/garbage_collector.go

```diff
@@ -347,7 +341,7 @@ var _ = SIGDescribe("Garbage collector", func() {
 		}
 		ginkgo.By("wait for all pods to be garbage collected")
 		// wait for the RCs and Pods to reach the expected numbers.
-		if err := wait.PollWithContext(ctx, 5*time.Second, (60*time.Second)+gcInformerResyncRetryTimeout, func(ctx context.Context) (bool, error) {
+		if err := wait.PollWithContext(ctx, f.Timeouts.Poll, (60*time.Second)+gcInformerResyncRetryTimeout, func(ctx context.Context) (bool, error) {
```

Comment:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/116724/pull-kubernetes-verify-strict-lint/1637178121083949056

> test/e2e/apimachinery/garbage_collector.go:344:13: SA1019: wait.PollWithContext is deprecated: This method does not return errors from context, use PollWithContextTimeout. Note that the new method will no longer return ErrWaitTimeout and instead return errors defined by the context package. Will be removed in a future release. (staticcheck)

This can be ignored at the moment, but perhaps now would be a good time to replace the call as the line is getting touched?

Reply:
I'm hoping to get this PR backported, and it looks like `PollWithContextTimeout` wasn't added until v1.27. I'll address this in a follow-up cleanup PR that doesn't need backporting (although I'll probably copy the `gomega.Eventually` pattern instead).